### PR TITLE
Allow `Annotated` and `ReadOnly` types for `TypedDict` keys

### PR DESCRIPTION
**Pull request by sobolevn**
Let's see how it goes 🚄 

Closes https://github.com/HypothesisWorks/hypothesis/issues/4071

### CODE REVIEW

**Comment by sobolevn** [comment]
3.8 is failing: https://github.com/HypothesisWorks/hypothesis/actions/runs/10262861146/job/28393427004?pr=4072 Fixing!

File: hypothesis-python/src/hypothesis/strategies/_internal/types.py
Comment (Zac-HD) [inline hypothesis-python/src/hypothesis/strategies/_internal/types.py:154]: Actually added in 3.13, per https://peps.python.org/pep-0705/

File: hypothesis-python/src/hypothesis/strategies/_internal/core.py
Comment (Zac-HD) [inline hypothesis-python/src/hypothesis/strategies/_internal/core.py:1401]: Is it really impossible to write a covering test for this?  If so, let's say why.

I'd also like to show the `get_pretty_function_description()` of the types in the error message.

File: hypothesis-python/src/hypothesis/strategies/_internal/core.py
Comment (Zac-HD) [inline hypothesis-python/src/hypothesis/strategies/_internal/core.py:1384]: I don't see a subsequent use of `v` here, maybe we can simplify the function?  I'd also be keen for a more verbose variable name than `qs`, it's hard enough for new contributors to read this file already 😅

File: hypothesis-python/RELEASE.rst
Comment (Zac-HD) [inline hypothesis-python/RELEASE.rst:4]: Let's use sphinx cross-references for these names, and explicitly mention that it's _in `from_type()`_.

**Review by Zac-HD — COMMENTED** [review]
Nice!  Probably just one round of comments before we merge:

File: hypothesis-python/src/hypothesis/strategies/_internal/types.py
Comment (sobolevn) [reply hypothesis-python/src/hypothesis/strategies/_internal/types.py:154]: Oups, the funny part is that I added it 😆

File: hypothesis-python/RELEASE.rst
Comment (sobolevn) [reply hypothesis-python/RELEASE.rst:4]: Sure, will do soon!

File: hypothesis-python/src/hypothesis/strategies/_internal/core.py
Comment (sobolevn) [reply hypothesis-python/src/hypothesis/strategies/_internal/core.py:1384]: Done! However, `v` is used in:

```python
anns[k] = from_type_guarded(v)
if anns[k] is ...:
     anns[k] = _from_type_deferred(v)
```

File: hypothesis-python/src/hypothesis/strategies/_internal/core.py
Comment (sobolevn) [reply hypothesis-python/src/hypothesis/strategies/_internal/core.py:1401]: Sorry, not quite sure about the `get_pretty_function_description` part. These are two set objects. They are not supported by `get_pretty_function_description` 🤔

Comment (Zac-HD) [reply hypothesis-python/src/hypothesis/strategies/_internal/core.py]: I meant for the typeddict, using `get_pretty_function_description(thing)` to represent it.  But since we think this is unreachable anyway it doesn't matter much 🙂

File: hypothesis-python/RELEASE.rst
Comment (Zac-HD) [inline hypothesis-python/RELEASE.rst:4]:
```suggestion
This release allows using :obj:`~python:typing.Annotated`
and :obj:`!ReadOnly` types
```
for consistent style when rendered

**Review by Zac-HD — APPROVED** [review]
Looks great!  Will merge once you (or I when I get home) tweak the changelog markup

**Comment by Zac-HD** [comment]
[beautiful!](https://hypothesis--4072.org.readthedocs.build/en/4072/changes.html#current-pull-request)

**Comment by sobolevn** [comment]
Thanks for the help! 👍 😊 🎉

**Pull request merged**
